src/bindings/python/flux/cli/base.py

```diff
@@ -757,7 +758,11 @@ def __init__(self, prog, usage=None, description=None, exclude_io=False):
         self.exitcode = 0
         self.progress = None
         self.watcher = None
+        self.plugins = CLIPluginRegistry().load_plugins(prog)
```

Comment:
@grondo What would you think of instead

```
self.plugins = CLIPluginRegistry(prog)
```
and the registry calling `load_plugins` as part of init? (or doing the same things but eliminating that function)

Reply:
We'd have to use `__new__` instead of `__init__` to return an instance of the class, perhaps that's why you went with the `load_plugins()` function instead in the first prototype.

Reply:
Ohh, I see why now. This is useful as-is in the validator. Ignore my noise!

Reply:
You make a good point, though. Since it is internal use only, we could eventually update the class initializer to work in the way you propose if someone thinks of a clean solution.

Reply:
I think I might have figured out how to do this, but I need to test it first.

Reply:
It does seem like you could just call `self.load_plugins(prog)` in the `__init__` function, once `prog` is added as an arg.

I guess it is just a style question at this point. The one time you may not want to do this is when you might add other setup or configuration methods to the class in the future. Then you want to be able to instantiate an object of the class _before_ loading plugins. This kind of reason is why I tend to first create classes that only set initial attribute values in a class initializer, and other methods to fully set up or configure the class.